rero_ils/modules/ebooks/utils.py

```diff
@@ -128,19 +128,26 @@ def update_document_holding(record, pid):
             item_type_pid = org.online_circulation_category()
             locations = org.get_online_locations()
             for location_pid in locations:
-                if not get_standard_holding_pid_by_doc_location_item_type(
-                    new_record.pid, location_pid, item_type_pid
+                if not get_holding_pid_by_doc_location_item_type(
+                    new_record.pid, location_pid, item_type_pid, 'electronic'
                 ):
                     create_holding(
                         document_pid=new_record.pid,
                         location_pid=location_pid,
                         item_type_pid=item_type_pid,
                         electronic_location=harvested_source,
-                        holdings_type='electronic')
-                holdings = get_holdings_by_document_item_type(
-                    new_record.pid, item_type_pid)
-                for holding in holdings:
-                    if holding.location_pid not in locations:
-                        holding.delete(
-                            force=False, dbcommit=True, delindex=True)
+                        holdings_type='electronic'
+                    )
+
+    source_uris = []
+    for harvested_source in harvested_sources:
```

Comment:
you can use filter: https://book.pythontips.com/en/latest/map_filter.html#filter